### PR TITLE
Make contracthashtool compatible with new libsecp256k1 API

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

File: contracthashtool.c [hunk 1]
```diff
@@ -170,7 +170,7 @@ int main(int argc, char* argv[]) {
 	if (mode == 0x2 && !priv_key_str)
 		USAGEEXIT("No private key specified\n");
 
-	secp256k1_context_t *secp256k1_ctx = secp256k1_context_create((unsigned int) -1);
+	secp256k1_context_t *secp256k1_ctx = secp256k1_context_create(SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
```
Comment (TheBlueMatt) [inline contracthashtool.c]: Can we get the secp256k1 header to specify which flags are needed for exactly which functions? The reason I went with -1 is that it is not at all obvious which context flags are needed for things like pubkey_tweak_add without reading the code.

File: contracthashtool.c [hunk 2]
```diff
@@ -242,12 +242,17 @@ int main(int argc, char* argv[]) {
 			for (i = 0; i < key_count; i++) {
 				unsigned char res[32];
 				hmac_sha256(res, keys_work[i], data, 4 + 16 + (ascii_contract ? strlen(ascii_contract) : 20));
-				
-				if (secp256k1_ec_pubkey_tweak_add(secp256k1_ctx, keys_work[i], 33, res) == 0) {
+				secp256k1_pubkey_t pubkey;
+				if (!secp256k1_ec_pubkey_parse(secp256k1_ctx, &pubkey, keys_work[i], 33)) {
```
Comment (TheBlueMatt) [inline contracthashtool.c]: Nit: coding style does not include braces for one-line if statements.

File: contracthashtool.c [hunk 3]
```diff
@@ -287,8 +293,9 @@ int main(int argc, char* argv[]) {
 		    memcpy(data + 4 + sizeof(nonce), p2sh_bytes,     sizeof(p2sh_bytes));
 
 		int len = 0;
-		if (secp256k1_ec_pubkey_create(secp256k1_ctx, pub, &len, priv, 1) != 1 || len != 33)
+		if (secp256k1_ec_pubkey_create(secp256k1_ctx, &pubkey, priv) != 1)
 			ERROREXIT("Private key was invalid\n");
+		secp256k1_ec_pubkey_serialize(secp256k1_ctx, pub, &len, &pubkey, 1);
```
Comment (TheBlueMatt) [inline contracthashtool.c]: Looks like you removed the length == 33 check?

File: contracthashtool.c (hunk 1 above)
Comment (sipa) [inline contracthashtool.c]: The pubkey_tweak functions need VERIFY. I'll update the documentation for that indeed.

File: contracthashtool.c (hunk 3 above)
Comment (sipa) [inline contracthashtool.c]: Fixed (and above as well), with an assert.

File: contracthashtool.c (hunk 2 above)
Comment (sipa) [inline contracthashtool.c]: Fixed.

**Pull request merged**